packages/vite/src/node/config.ts

```diff
@@ -663,7 +663,7 @@ export async function resolveConfig(
       ...config.experimental
     }
   }
-  const resolved: ResolvedConfig = Object.assign(config, resolvedConfig)
+  const resolved: ResolvedConfig = Object.assign({}, config, resolvedConfig)
```

Comment:
Maybe it is better to use object spread syntax if we are not modifying the first argument?